guide/adoc/portfile-keywords.adoc

```diff
@@ -219,4 +219,21 @@ license_noconflict openssl
 [source]
 ----
 license_noconflict readline gdbm
+----
+use_xcode::
+Available in 2.6.0 and later. By default, it is assumed on ``macosx`` that ports will not need tools from Xcode.app unless (1) Command Line Tools aren't installed, (2) you are on an old version of Mac OS X that does not support the xcode-select mechanism, or (3) the port uses build.type xcode or is in the xcode PortGroup.
+If a port needs to use Xcode (i.e., xcodebuild) in any way, ``use_xcode yes`` should be set or the port should be included in the xcode PortGroup.
+We now export DEVELOPER_DIR during all build phases, which means libxcselect shims (i.e., /usr/bin/clang) will resolve to CLT or Xcode depending on use_xcode. Build systems that ignore the environment may accidentally use Xcode which will cause a failure in trace mode.
```

Comment:
I'd rewrite this to a passive form, e.g. "The environment variable `DEVELOPER_DIR` is now exported during all build phases...". I would probably be good to also note that `DEVELOPER_DIR` is set to the value of `${configure.developer_dir}` and that option's default automatically changes depending on `use_xcode`.